src/store/model-store.js

```diff
@@ -75,4 +85,8 @@ store.registerHandler('NEW_MODEL', data => {
     models.push(model);
 });
 
+store.registerHandler('DELETE_MODEL', data => {
+    store.deleteModelByAddress(data.modelAddress);
```

Comment:
Yes to `store.emitChange()`.

Also, there should be no `store.deleteModelByAddress` method, see above. Move `deleteModelAddress` to a function outside of the store (but still in this file).